### DISCUSSION TITLE
Add README.md to every Directus `/extensions/*` subdirectory to preserve directory if empty in Git

### DISCUSSION THREAD

**Discussion by Anoesj** [directus/directus#15980] [Feature Requests]
Hey there!

I first installed Directus last month and one thing I noticed is that when creating an app with `pnpm create directus-app my-project`, an `/extensions` directory is created, but all subdirectories (`/hooks`, `/endpoints` etc.) are empty. This makes Git ignore these directories, because they are empty.

What would help for easier collaboration is if every single one of those subdirectories would contain a `README.md` file linking to the documentation of that kind of extension. This would make Git commit those `README.md` files, so the extension subdirectories are preserved and instantly available for collaborators when cloning a Directus application. Adding a README seems better than adding a `.gitkeep` file, which is not actually a feature of Git.

How do you feel about this? 😄

**Comment by br41nslug** [comment]
Good suggestion, it couldn't hurt to add such files 😄I used to do that often in PHP projects with placeholder files called `empty`

**Comment by rijkvanzanten** [comment]
Orrrrrrrr we could simply not auto-generate those folders if they don't exist. Those folders aren't really necessary to run Directus at all 🤷🏻

**Reply by Anoesj** [reply]
I'm quite new to Directus and I actually really liked how they were auto-generated! 😬

**Reply by Anoesj** [reply]
In fact, my first experience with Directus is that is is amazing, but extending it is more difficult than it could be. I would love to be able to skip the whole Rollup/make-it-an-npm-package part and just extend Directus directly (ha).

**Reply by br41nslug** [reply]
Having people make these folders manually when they need them for local extensions sounds like it would decrease DX/UX to me 😬

**Reply by rijkvanzanten** [reply]
True!

**Reply by nickrum** [reply]
We had a few reports from users having issues with Directus trying to ensure that those folders are present at runtime. I think a good compromise would be to auto-generate the folders when using `create-directus-project`, but simply ignoring the absence of them at runtime.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
